mdd/python/translation/openconfig_xe/xe_qos.py

```diff
@@ -0,0 +1,281 @@
+    device_int = device_cdb.ios__interface
+    if interface_type == 'GigabitEthernet':
+        device_int.GigabitEthernet.create(interface_number)
+        device_int.GigabitEthernet[interface_number].service_policy.output = sequence.output.config.output_fwd_group
+    elif interface_type == 'Port_channel':
+        device_int.Port_channel.create(interface_number)
+        device_int.Port_channel[interface_number].service_policy.output = sequence.output.config.output_fwd_group
+    elif interface_type == 'Loopback':
+        device_int.Loopback.create(interface_number)
+        if sequence.one_rate_two_color.config.queuing_behavior == 'POLICE' or sequence.two_rate_three_color:
+            device_int.Loopback[interface_number].service_policy.output = sequence.output.config.output_fwd_group
+    elif interface_type == 'Tunnel':
+        device_int.Tunnel.create(interface_number)
+        device_int.Tunnel[interface_number].service_policy.output = sequence.output.config.output_fwd_group
+    else:
+        raise ValueError(
+            f'Interface type {interface_type} not supported by this NSO_OC_Services implementation. Please file an issue at https://github.com/model-driven-devops/nso-oc-services')
+
+def conf_in_service_policy(device_cdb, interface, sequence):
+
+    interface_type, interface_number = get_interface_type_and_number(interface.config.interface_id)
+
+    # Configure service-policy-input
+    device_int = device_cdb.ios__interface
+    if interface_type == 'GigabitEthernet':
+        device_int.GigabitEthernet.create(interface_number)
+        if sequence.one_rate_two_color.config.queuing_behavior == 'POLICE' or sequence.two_rate_three_color:
+            device_int.GigabitEthernet[interface_number].service_policy.input = sequence.output.config.output_fwd_group
+    elif interface_type == 'Port_channel':
+        device_int.Port_channel.create(interface_number)
+        device_int.Port_channel[interface_number].service_policy.input = sequence.output.config.output_fwd_group
+    elif interface_type == 'Loopback':
+        device_int.Loopback.create(interface_number)
+        if sequence.one_rate_two_color.config.queuing_behavior == 'POLICE' or sequence.two_rate_three_color:
+            device_int.Loopback[interface_number].service_policy.input = sequence.output.config.output_fwd_group
+    elif interface_type == 'Tunnel':
+        device_int.Tunnel.create(interface_number)
+        if sequence.one_rate_two_color.config.queuing_behavior == 'POLICE' or sequence.two_rate_three_color:
+            device_int.Tunnel[interface_number].service_policy.input = sequence.output.config.output_fwd_group
+    else:
+        raise ValueError(
+            f'Interface type {interface_type} not supported by this NSO_OC_Services implementation. Please file an issue at https://github.com/model-driven-devops/nso-oc-services')
+
+def modify_dscp(dscp):
+    if (dscp % 2) != 0:
+        return dscp
+    if dscp in dscp_dict.keys():
```

Comment:
This if statement isn't required. The get function will perform a check if that key exist, otherwise that second arg ('default') value is returned.

Reply:
If statement removed.